exercise-integration-testing-5-53/test/🏅-your-mission-test-5-53.spec.js

```diff
@@ -0,0 +1,148 @@
+// 🏅 Your mission is to create your first integration tests here 💜
+// ✅ Whenever you see this icon, there's a TASK for you
+// 💡 - This is an ADVICE symbol, it will appear nearby most tasks and help you in fulfilling the tasks
+
+
+const { initializeAPI, closeApi} = require("../sensors-api");
+const nock = require("nock");
+const supertest = require("supertest");
+
+let expressApp;
+
+beforeAll(async () => {
+  expressApp = await initializeAPI();
+});
+
+
+let mockedNotifications;
+
+beforeEach(() => {
+  // 📗 Reading exercise: Why is this needed 👇? Read about npm/nock
+  mockedNotifications=nock("http://localhost").get("/notification").reply(200, {
+    success: true,
+  });
+});
+
+afterEach(()=>{
+  nock.cleanAll();
+})
+
+// ✅ Ensure that the webserver is closed when all the tests are completed
+// 💡 TIP: Use the right test hook to call the API and instruct it to close
+afterAll(async ()=>{
+  await closeApi();
+})
+
+describe('/api',()=>{
+  describe("/sensor-events", () => {
+    // ✅ TASK: Test that when a new event is posted to /event route, if category or temperature are not specified -> the API returns HTTP 400
+    // 💡 TIP: Down below, there is an example event schema
+    describe("/post",()=> {
+      test("When category is not specified, should get http 400 error", async () => {
+        //Arrange
+        const eventToAdd = {
+          temperature: 20,
+          name: "Thermostat-temperature", //This must be unique
+          color: "Green",
+          weight: "80 gram",
+          status: "active",
+          // 💡 TIP: Consider explicitly specify that category is undefined
+        };
+
+        //Act
+        // 💡 TIP: use any http client lib like Axios OR supertest
+        const receivedResponse = await supertest(expressApp).post("/sensor-events").send(eventToAdd);
+        // 💡 TIP: This is how it is done with Supertest -> await request(expressApp).post("/sensor-events").send(eventToAdd);
+
+
+        //Assert
+        // 💡 TIP: verify that status is 400
+        expect(receivedResponse.status).toBe(400)
+      });
+      // ✅ TASK: Test that when a new event is posted to /sensor-events route, the temperature is not specified -> the event is NOT saved to the DB!
+      // 💡 TIP: Testing the response is not enough, the adequate state (e.g. DB) should also satisfy the expectation
+      it("when temp is no defined the event is not save to db", async () => {
+        //Arrange
+        const eventToAdd = {
+          temperature: undefined,
+          name: "no-temp-test",
+          color: "Green",
+          weight: "80 gram",
+          status: "active",
+          category: "sensor"
+        };
+
+        //Act
+        await supertest(expressApp).post("/sensor-events").send(eventToAdd);
+        const pullFromDb = await supertest(expressApp).get("/sensor-events/no-temp-test/name");
```

Comment:
Not sure, but might be a bug here: You use the event.name within the path instead of event.category

How could you realize this: Always change your code under test and make the tests fail (e.g., comment the validation part). I suspect that this test will never fail, am I correct?

Reply:
you are correct
but in this specific case i think even if i changed the code slightly the code will pass due to its working in a negative way (not adding the event) maybe when working to check if something was *not* added should add it explicitly in the code and then the tests should fail